lib/validate/validate_object.js

```diff
@@ -35,8 +35,7 @@ module.exports = function validateObject(options) {
             }));
 
         // tolerate root-level extra keys & arbitrary layer properties
-        // TODO remove this layer-specific logic
-        } else if (key !== '' && key.split('.').length !== 1) {
+        } else if (key !== '' && !options.allowArbitraryKeys) {
```

Comment:
Do we need the `key !== ''` part of this conditional? It looks like that's covered by passing `allowArbitraryKeys: true` in `validateStyleMin`.